### PR TITLE
Trim <bsd.cheri.mk>

### PR DESCRIPTION
**Pull request by bsdjhb**
This doesn't completely remove the file, but this does put it on a diet towards the goal of removing it.

### CODE REVIEW

File: share/mk/bsd.cheri.mk [hunk 1]
```diff
@@ -50,32 +50,8 @@ NEED_CHERI=	pure
...
-# to use 32 bytes anyway
-CFLAGS+=	-mstack-alignment=16
-.endif # $COMPILER_TYPE == clang
-.endif # MIPS, not hybrid (adjust stack alignment)
-
 .if ${MK_CHERI} != "no" && defined(WANT_CHERI) && ${WANT_CHERI} != "none"
-_CHERI_COMMON_FLAGS=	-integrated-as --target=mips64-unknown-freebsd \
-			-msoft-float \
-			-cheri-uintcap=${CHERI_UINTCAP_MODE:Uaddr}
+_CHERI_COMMON_FLAGS=	-cheri-uintcap=${CHERI_UINTCAP_MODE:Uaddr}
```
Comment (bsdjhb) [inline share/mk/bsd.cheri.mk:54]: I wonder if we can kill this.  The only user currently is usr.bin/truss/Makefile which sets it to 'addr' explicitly.

File: share/mk/bsd.compat.mk [hunk 2]
```diff
@@ -157,9 +157,13 @@ LIB64_MACHINE_ABI=	${MACHINE_ABI:Npurecap}
 .endif
 HAS_COMPAT+=CHERI
 LIBCHERICPUFLAGS=  -target mips64-unknown-freebsd13.0 -cheri -mabi=purecap
+LIBCHERICPUFLAGS+=	-fpic
+LIBCHERICPUFLAGS+=	-Qunused-arguments
```
Comment (arichardson) [inline share/mk/bsd.compat.mk:161]: Is this still needed? I feel like I fixed most flags to be marked as consumed in clang.

File: share/mk/bsd.compat.mk [hunk 3]
```diff
@@ -157,9 +157,13 @@ LIB64_MACHINE_ABI=	${MACHINE_ABI:Npurecap}
...
+LIBCHERICPUFLAGS+=	-Werror=cheri-bitwise-operations
 LIBCHERI_MACHINE=	mips
 LIBCHERI_MACHINE_ARCH=	mips64c128
-
+LIBCHERILDFLAGS=	-fuse-ld=lld
+LIBCHERILDFLAGS+=	-Wl,-preemptible-caprelocs=elf
```
Comment (arichardson) [inline share/mk/bsd.compat.mk:166]: Should be the default so we can drop this.

File: share/mk/bsd.cpu.mk [hunk 4]
```diff
@@ -343,11 +342,12 @@ CFLAGS+=	-Werror=implicit-function-declaration
...
-CFLAGS+=	-cheri=128
+CFLAGS+=	-cheri
 LDFLAGS+=	-fuse-ld=lld
-# XXXBD: still needed?
-LDFLAGS+=	-Wl,-melf64btsmip_cheri_fbsd
 LDFLAGS+=	-Wl,-preemptible-caprelocs=elf
+
+# XXX: Needed as Clang rejects -mllvm -cheri128 when using $CC to link:
+# warning: argument unused during compilation: '-cheri=128'
 CFLAGS+=	-Qunused-arguments
```
Comment (arichardson) [inline share/mk/bsd.cpu.mk:351]: Again I believe this has been fixed.

File: share/mk/bsd.cpu.mk [hunk 5]
```diff
@@ -343,11 +342,12 @@ CFLAGS+=	-Werror=implicit-function-declaration
 # Turn off deprecated warnings
 # XXXBD: is this still needed?
 CFLAGS+=	-Wno-deprecated-declarations
-CFLAGS+=	-cheri=128
+CFLAGS+=	-cheri
 LDFLAGS+=	-fuse-ld=lld
-# XXXBD: still needed?
-LDFLAGS+=	-Wl,-melf64btsmip_cheri_fbsd
 LDFLAGS+=	-Wl,-preemptible-caprelocs=elf
```
Comment (arichardson) [inline share/mk/bsd.cpu.mk:347]: Can also be removed.

File: share/mk/bsd.cpu.mk [hunk 6]
```diff
@@ -343,11 +342,12 @@ CFLAGS+=	-Werror=implicit-function-declaration
 # Turn off deprecated warnings
 # XXXBD: is this still needed?
 CFLAGS+=	-Wno-deprecated-declarations
```
Comment (arichardson) [inline share/mk/bsd.cpu.mk:344]:
```suggestion
```
I think this was only needed for deprecated `__capability` placement and I (think I) fixed all of those ~2 years ago.

Comment (jrtc27) [reply share/mk/bsd.cpu.mk:344]: It's under `mips*c*` so I'd hope so...

File: share/mk/bsd.compat.mk (hunk 2 above)
Comment (bsdjhb) [reply share/mk/bsd.compat.mk:161]: I do still get warnings, but they aren't fatal.  They scrolled off my terminal, but I was seeing warnings about unused "-L" passed to clang-cpp during the libcheri build of a hybrid world.  I will keep this removed though if my world finishes without errors.

File: share/mk/bsd.cheri.mk (hunk 1 above)
Comment (bsdjhb) [reply share/mk/bsd.cheri.mk:54]: Done.

File: share/mk/bsd.compat.mk (hunk 3 above)
Comment (bsdjhb) [reply share/mk/bsd.compat.mk:166]: Done (in both places)

File: share/mk/bsd.cpu.mk (hunk 6 above)
Comment (bsdjhb) [reply share/mk/bsd.cpu.mk:344]: Done.

**Pull request merged**